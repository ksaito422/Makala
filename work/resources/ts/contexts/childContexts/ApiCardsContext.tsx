import React, { useState, createContext, useContext } from 'react';
import { FeedbackContext } from './FeedbackContext';
import { instance } from '../../config/axios.config';

export const ApiCardsContext = createContext({});

export const ApiCardsContextProvider: React.FC = (props) => {
  /**
   * { api通信中のスピナー表示のon/off管理, api通信の結果通知 }
   * { items: ボードアイテムで表示するデータ, numberMade: 今までにカードを作った総数 }
   */
  const { setProgress, setStatus } = useContext<any>(FeedbackContext);
  const [cardsState, setCardsState] = useState([]);

  // カードを取得するapiと通信
  const getCards = async (board_name: string) => {
    // スピナーon
    // トークン取得
    await setProgress(true);
    const token = localStorage.getItem('makala_token');

    await instance({
      method: 'GET',
      url: `api/v1/cards/${board_name}`,
      headers: {
        'Authorization': `Bearer ${token}`
      }
    })
    .then((res) => {
      // 無事データを取得できたら、cardsStateに保管
      setCardsState(res.data.cards);
      return;
    })
    .catch(() => {
      setStatus({
        open: true,
        type: 'error',
        message: 'データの取得に失敗しました。'
      });
      return;
    })
    .finally(() => {
      // スピナーoff
      setProgress(false);
      return;
    })
  }

  // カードを新規作成するapiと通信
  const createCard = async (
    card: any,
    data: {
      board_name: string,
      card_name: string,
      card_content: string,
    }) => {
      // スピナーon
      // トークン取得
      await setProgress(true);
      const token = localStorage.getItem('makala_token');

      await instance({
        method: 'POST',
        url: `api/v1/cards`,
        data: data,
        headers: {
          'Authorization': `Bearer ${token}`
        }
      })
      .then((res) => {
        setStatus({
          open: true,
          type: 'success',
          message: res.data.message
        });
        return;
      })
      .catch(() => {
        setStatus({
          open: true,
          type: 'error',
          message: 'カードの作成に失敗しました。'
        });
        return;
      })
      .finally(() => {
        setProgress(false);
        getCards(card);
        return;
      })
    }

  // カードを更新するロジックapiと通信
  const updateCard = async (
    id: number,
    card: any,
    data: {
      card_name: string,
      card_content: string,
    }) => {
      // スピナーon
      // トークン取得
      await setProgress(true);
      const token = localStorage.getItem('makala_token');

      await instance({
        method: 'PUT',
        url: `api/v1/cards/${id}`,
        data: data,
        headers: {
          'Authorization': `Bearer ${token}`
        }
      })
      .then((res) => {
        setStatus({
          open: true,
          type: 'success',
          message: res.data.message
        });
        return;
      })
      .catch(() => {
        setStatus({
          open: true,
          type: 'error',
          message: 'カードの内容変更に失敗しました。'
        });
        return;
      })
      .finally(() => {
        // スピナーoff
        setProgress(false);
        getCards(card);
        return;
      })
    }

  // カードを削除するapiと通信
  const deleteCard = async (id: number, card: any) => {
    // スピナーon
    // トークン取得
    await setProgress(true);
    const token = localStorage.getItem('makala_token');

    await instance({
      method: 'DELETE',
      url: `api/v1/cards/${id}`,
      headers: {
        'Authorization': `Bearer ${token}`
      }
    })
    .then((res) => {
      setStatus({
        open: true,
        type: 'success',
        message: res.data.message
      });
      return;
    })
    .catch(() => {
      setStatus({
        open: true,
        type: 'error',
        message: 'カードの削除に失敗しました。'
      });
      return;
    })
    .finally(() => {
      // スピナーoff
      setProgress(false);
      getCards(card);
      return;
    })
  }

  return (
    <ApiCardsContext.Provider value={{
      cardsState,
      setCardsState,
      getCards,
      createCard,
      updateCard,
      deleteCard
      }}
    >
      {props.children}
    </ApiCardsContext.Provider>
  );
}