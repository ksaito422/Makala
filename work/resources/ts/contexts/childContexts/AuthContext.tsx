import React, { useState, createContext, useContext } from 'react';
import { FeedbackContext } from './FeedbackContext';
import { instance } from '../../config/axios.config';

type AuthUserState = {
    id: number | null,
    name: string | null,
}

export const AuthContext = createContext({});

export const AuthContextProvider: React.FC = (props) => {
  /**
   * { api通信中のスピナー表示のon/off管理, api通信の結果通知 }
   * 認証情報を保持するstate
   */
  const { setProgress, setStatus } = useContext<any>(FeedbackContext);
  const [authUserState, setAuthUserState] = useState<AuthUserState>({
    id: null,
    name: null,
  });

  // ログイン状態の管理
  // 認証成功したらisAuthをtrueにする
  // ログアウトしたらisAuthをfalseにする
  const [isAuth, setIsAuth] = useState(false);
  const login = () => {
    setIsAuth(true);
  }
  const logout = () => {
    setIsAuth(false);
  }

  // ログイン処理を行うapiと通信
  const authLogin = async (
    data: {
      'email': string,
      'password': string,
  }) => {
    // スピナーon
    await setProgress(true);

    await instance({
      method: 'POST',
      url: 'api/auth/login',
      data: data,
    })
    .then((res) => {
      // ローカルストレージに認証情報を保管 *脆弱性のことはあとで考える
      // set isAuth to true
      // 通信結果の通知内容
      localStorage.setItem('makala_token', res.data.access_token);
      localStorage.setItem('makala_user', res.data.name);
      setAuthUserState({ ...authUserState, id: res.data.id, name: res.data.name});
      login();
      setStatus({
        open: true,
        type: 'success',
        message: 'ログインしました。'
      });
      return;
    })
    .catch(() => {
      // 通信結果の通知内容
      setStatus({
        open: true,
        type: 'error',
        message: 'ログインに失敗しました。'
      });
      return;
    })
    .finally(() => {
      // スピナーoff
      setProgress(false);
      return;
    });
  }

  // ユーザー登録を行うapiと通信
  const authRegister = async (
    data: {
      'name': string,
      'email': string,
      'password': string,
  }) => {
    // スピナーon
    await setProgress(true);

    await instance({
      method: 'POST',
      url: 'api/auth/register',
      data: data,
    })
    .then((res) => {
      // ローカルストレージに認証情報を保管 *脆弱性のことはあとで考える
      // set isAuth to true
      // 通信結果の通知内容
      localStorage.setItem('makala_token', res.data.access_token);
      localStorage.setItem('makala_user', res.data.name);
      setAuthUserState({ ...authUserState, id: res.data.id, name: res.data.name});
      login();
      setStatus({
        open: true,
        type: 'success',
        message: 'ユーザー登録が完了しました。'
      });
      return;
    })
    .catch(() => {
      // 通信結果の通知内容
      setStatus({
        open: true,
        type: 'error',
        message: 'ユーザー登録に失敗しました。'
      });
      return;
    })
    .finally(() => {
      // スピナーoff
      setProgress(false);
      return;
    });
  }

  // ログアウト処理を行うapiと通信
  const authLogout = async () => {
    // スピナーon
    // トークン取得
    await setProgress(true);
    const token = localStorage.getItem('makala_token');

    await instance({
      method: 'POST',
      url: 'api/auth/logout',
      headers: {
        'Authorization': `Bearer ${token}`
      }
    })
    .then((res) => {
      // ローカルストレージの認証情報を削除
      // set isAuth to false
      // 通信結果の通知内容
      localStorage.removeItem('makala_token');
      localStorage.removeItem('makala_user');
      logout();
      setStatus({
        open: true,
        type: 'success',
        message: res.data.message
      });
      return;
    })
    .catch(() => {
      // 通信結果の通知内容
      setStatus({
        open: true,
        type: 'error',
        message: 'ログアウトに失敗しました。'
      });
      return;
    })
    .finally(() => {
      // スピナーoff
      setProgress(false);
      return;
    });
  }

  // トークンリフレッシュするapiと通信
  const authRefresh = async () => {
    // トークン取得
    const token = localStorage.getItem('makala_token');

    await instance({
      method: 'POST',
      url: 'api/auth/refresh',
      headers: {
        'Authorization': `Bearer ${token}`
      }
    })
    .then((res) => {
      // 再度取得した、トークンをセット
      localStorage.setItem('makala_token', res.data.access_token);
      return;
    })
    .catch(() => {
      return;
    })
  }

  // ユーザー情報を取得するapiと通信
  const authMe = async () => {
    // スピナーon
    // トークン取得
    await setProgress(true);
    const token = localStorage.getItem('makala_token');

    await instance({
      method: 'POST',
      url: 'api/auth/me',
      headers: {
        'Authorization': `Bearer ${token}`
      }
    })
    .then((res) => {
      // ユーザー名とidをセット
      // isAuthをtrueにする
      // トークンリフレッシュする
      localStorage.setItem('makala_user', res.data.name);
      setAuthUserState({ ...authUserState, id: res.data.id, name: res.data.name});
      login();
      authRefresh();
      return;
    })
    .catch(() => {
      return;
    })
    .finally(() => {
      // スピナーoff
      setProgress(false);
      return;
    })
  }

  // ゲストログイン処理を行うapiと通信
  const authGuestLogin = async () => {
    // スピナーon
    // ゲストアカウントの認証情報
    await setProgress(true);
    const data = {
      'email': 'guest@example.com',
      'password': 'guest1234'
    }

    await instance({
      method: 'POST',
      url: 'api/auth/login',
      data: data,
    })
    .then((res) => {
      // ローカルストレージに認証情報を保管 *脆弱性のことはあとで考える
      // set isAuth to true
      // 通信結果の通知内容
      localStorage.setItem('makala_token', res.data.access_token);
      localStorage.setItem('makala_user', res.data.name);
      setAuthUserState({ ...authUserState, id: res.data.id, name: res.data.name});
      login();
      setStatus({
        open: true,
        type: 'success',
        message: 'ゲストログインしました。'
      });
      return;
    })
    .catch(() => {
      // 通信結果の通知内容
      setStatus({
        open: true,
        type: 'error',
        message: 'ログインに失敗しました。'
      });
      return;
    })
    .finally(() => {
      // スピナーoff
      setProgress(false);
      return;
    });
  }

  return (
    <AuthContext.Provider value={{
      authUserState,
      setAuthUserState,
      isAuth,
      login,
      authLogin,
      authRegister,
      authLogout,
      authMe,
      authGuestLogin,
      }}
    >
      {props.children}
    </AuthContext.Provider>
  );
}