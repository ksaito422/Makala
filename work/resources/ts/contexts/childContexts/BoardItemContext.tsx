import React, { useState, createContext } from 'react';

export const BoardItemContext = createContext({});

type ItemType = {
  id: string;
  content: string;
}

// BoardItemに渡すデータ
export const BoardItemContextProvider: React.FC = props => {
  const initial: ItemType[] = Array.from({ length: 5 }, (v, k) => k).map(k => {
    return {
      id: `${k}`,
      title: `id-${k}`,
      content: `Item ${k}`
    }
  })

  // { items: ボードアイテムで表示するデータ, numberMade: 今までにカードを作った総数 }
  const [BoardItemState, setBoardItemState] = useState({ items: initial, numberMade: initial.length });

  return (
    <BoardItemContext.Provider value={{BoardItemState, setBoardItemState}}>
      {props.children}
    </BoardItemContext.Provider>
  )
}