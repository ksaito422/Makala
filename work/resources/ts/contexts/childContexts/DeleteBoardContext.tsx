import React, { createContext } from 'react';
import axios from 'axios';

export const DeleteBoardContext = createContext({});

export const DeleteBoardContextProvider: React.FC = props => {
  // apiと通信して、ボードを削除するロジック
  const deleteBoard: any = (id: number) => {
    axios({
      method: 'DELETE',
      url: `/api/v1/boards/${id}`,
      headers: {
        'Content-Type': 'application/json',
      }
    })
    .then((res) => {
      console.log(res.data.message);
    })
    .catch((err) => {
      return;
    })
  }

  return (
    <DeleteBoardContext.Provider value={{deleteBoard}}>
      {props.children}
    </DeleteBoardContext.Provider>
  )
}