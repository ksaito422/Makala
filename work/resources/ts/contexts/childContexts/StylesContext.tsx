import React, { createContext } from 'react';
import { makeStyles } from '@material-ui/core';

export const StylesContext = createContext({});

// ページ全体で使うclassNameの定義
export const StylesContextProvider: React.FC = (props) => {
  const useStyles: any = makeStyles((theme) => ({
    // pagesで使用 メインコンテンツとヘッダーの間隔をとるため
    main_container: {
      marginTop: theme.spacing(4),
    },
    // dragBoradの高さとスクロール可否
    drop_able: {
      maxHeight: `70vh`,
      overflowY: 'scroll',
    },
    // Draggable内の要素の間隔をとるため
    drag_card: {
      marginBottom: theme.spacing(2),
    },
    // modalの表示サイズ
    modal: {
      position: 'absolute',
      width: `80%`,
      backgroundColor: theme.palette.background.paper,
      border: '2px solid #000',
      boxShadow: theme.shadows[5],
      padding: theme.spacing(2, 4, 3),
    },
    // スマホサイズのmodal表示サイズ
    modal_responsive: {
      position: 'absolute',
      width: `95%`,
      backgroundColor: theme.palette.background.paper,
      border: '2px solid #000',
      boxShadow: theme.shadows[5],
      padding: theme.spacing(2, 4, 3),
    },
    // アイコンを中央に
    centerPlacement: {
      textAlign: 'center',
    },
    // アイコンを右に
    rightPlacement: {
      textAlign: 'right',
    },
    // MarkdownPreview Paperのスタイル
    preview: {
      height: `70vh`,
      maxHeight: `70vh`,
      overflowY: 'scroll',
      padding: theme.spacing(2),
    },
    // AccodionAreaのgif画の高さ
    gif: {
      height: `60vh`,
    },
    // login formのスタイル
    auth: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
    },
    auth_form: {
      width: `100%`,
      marginTop: theme.spacing(4),
    },
    form_board: {
      width: `100%`,
    },
    // Boardsのボタンラベルに適用
    label: {
      justifyContent: `left`,
    },
    // スピナーに適用
    Spinner: {
      zIndex: 99,
    },
    // アバターの色
    avatar: {
      color: theme.palette.primary.dark,
      backgroundColor: theme.palette.primary.light,
    },
    // スマホ画面対応のアバターサイズ
    avatar_small: {
      width: `26px`,
      height: `26px`,
    },
    // aタグの装飾
    Link: {
      textDecoration: `none`,
      color: theme.palette.primary.main,
    },
    // cardのスタイル
    card: {
      justifyContent: 'left',
    },
    setting_contents: {
      marginTop: theme.spacing(4),
      padding: theme.spacing(2),
    },
  }));
  return <StylesContext.Provider value={{ useStyles }}>{props.children}</StylesContext.Provider>;
};
