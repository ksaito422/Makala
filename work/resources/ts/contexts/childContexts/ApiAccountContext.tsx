import React, { createContext, useContext } from 'react';
import { FeedbackContext } from './FeedbackContext';
import { AuthContext } from './AuthContext';
import { instance } from '../../config/axios.config';

export const ApiAccountContext = createContext({});

export const ApiAccountContextProvider: React.FC = (props) => {
  /**
   * { api通信中のスピナー表示のon/off管理, api通信の結果通知 }
   * ログインユーザーの情報
   * getBoardsで取得したデータを保管
   */
  const { setProgress, setStatus } = useContext<any>(FeedbackContext);
  const { authUserState, setAuthUserState } = useContext<any>(AuthContext);

  // ユーザー名を更新するapiと通信
  const changeName = async (name: string, userId: number) => {
    // スピナーon
    // トークン取得
    await setProgress(true);
    const token = localStorage.getItem('makala_token');

    await instance({
      method: 'PUT',
      url: `api/v1/account/name/${userId}`,
      data: { name },
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
      .then((res) => {
        // ローカルストレージのユーザー名を置換する
        // ユーザー情報を管理するstateのnameを更新する
        localStorage.setItem('makala_user', name);
        setAuthUserState({
          ...authUserState,
          name,
        });
        setStatus({
          open: true,
          type: 'success',
          message: res.data.message,
        });
      })
      .catch((err) => {
        setStatus({
          open: true,
          type: 'error',
          message: 'ユーザー名の変更に失敗しました。',
        });
      })
      .finally(() => {
        // スピナーoff
        setProgress(false);
      });
  };

  // メールアドレスを変更するapiと通信
  const changeEmail = async (newEmail: string, email: string, password: string, userId: number) => {
    // スピナーon
    // トークン取得
    await setProgress(true);
    const token = localStorage.getItem('makala_token');
    const data = { newEmail, email, password, userId };

    await instance({
      method: 'PUT',
      url: `api/v1/account/email/${userId}`,
      data,
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
      .then((res) => {
        // ユーザー情報を管理するstateのnameを更新する
        setAuthUserState({ ...authUserState, email: newEmail });
        setStatus({
          open: true,
          type: 'success',
          message: res.data.message,
        });
      })
      .catch((err) => {
        setStatus({
          open: true,
          type: 'error',
          message: err.response.data.message,
        });
      })
      .finally(() => {
        // スピナーoff
        setProgress(false);
      });
  };

  return (
    <ApiAccountContext.Provider value={{ changeName, changeEmail }}>
      {props.children}
    </ApiAccountContext.Provider>
  );
};
