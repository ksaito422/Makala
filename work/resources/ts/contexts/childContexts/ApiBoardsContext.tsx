import React, { useState, createContext, useContext } from 'react';
import { FeedbackContext } from './FeedbackContext';
import { AuthContext } from './AuthContext';
import axios from 'axios';

export const ApiBoardsContext = createContext({});

export const ApiBoardsContextProvider: React.FC = (props) => {
  /**
   * { api通信中のスピナー表示のon/off管理, api通信の結果通知 }
   * ログインユーザーの情報
   * getBoardsで取得したデータを保管
   */
  const { setProgress, setStatus } = useContext<any>(FeedbackContext);
  const { authUserState } = useContext<any>(AuthContext);
  const [boardsState, setBoardsState] = useState([]);

  // ボードを取得するapiと通信
  const getBoards = async () => {
    // スピナーon
    // トークン取得
    await setProgress(true);
    const token = localStorage.getItem('makala_token');

    await axios({
      method: 'GET',
      url: `/api/v1/boards/${authUserState.name}`,
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`
      }
    })
    .then((res) => {
      // 無事データを取得できたら、boardsStateに保管
      setBoardsState(res.data.boards);
      return;
    })
    .catch(() => {
      setStatus({
        open: true,
        type: 'error',
        message: 'データの取得に失敗しました。'
      });
      return;
    })
    .finally(() => {
      // スピナーoff
      setProgress(false);
      return;
    })
  }

  // ボードを作成するapiと通信
  const createBoard = async (
    data: {
      user_id: number,
      board_name: string,
    }) => {
      // スピナーon
      // トークン取得
      await setProgress(true);
      const token = localStorage.getItem('makala_token');

      await axios({
        method: 'POST',
        url: `/api/v1/boards`,
        data: data,
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${token}`
        }
      })
      .then((res) => {
        setStatus({
          open: true,
          type: 'success',
          message: res.data.message
        });
        return;
      })
      .catch(() => {
        setStatus({
          open: true,
          type: 'error',
          message: 'ボードの作成に失敗しました。'
        });
        return;
      })
      .finally(() => {
        // スピナーoff
        setProgress(false);
        getBoards();
        return;
      })
    }

  // ボード名を更新するapiと通信
  const updateBoard = async (
    data: {
      id: number,
      board_name: string,
    }) => {
      // スピナーon
      // トークン取得
      await setProgress(true);
      const token = localStorage.getItem('makala_token');

      await axios({
        method: 'PUT',
        url: `/api/v1/boards/${data.id}`,
        data: data.board_name,
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${token}`
        }
      })
      .then((res) => {
        setStatus({
          open: true,
          type: 'success',
          message: res.data.message
        });
        return;
      })
      .catch(() => {
        setStatus({
          open: true,
          type: 'error',
          message: 'ボード名の変更に失敗しました。'
        });
        return;
      })
      .finally(() => {
        // スピナーoff
        setProgress(false);
        getBoards();
        return;
      })
    }

  // パフォーマンス向上させるときに使う
  // ボードの更新時に、更新部分だけを再度stateにセットし直している
  const updateBoardState = (
    obj: {
      index: number,
      board_name: string
    }) => {
    const newBoardsState: any = [ ...boardsState ];
    newBoardsState[obj.index].board_name = obj.board_name;
    setBoardsState(newBoardsState);
  }

  // ボードを削除するapiと通信
  const deleteBoard = async (id: number) => {
    // スピナーon
    // トークン取得
    await setProgress(true);
    const token = localStorage.getItem('makala_token');

    await axios({
      method: 'DELETE',
      url: `/api/v1/boards/${id}`,
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`
      }
    })
    .then((res) => {
      setStatus({
        open: true,
        type: 'success',
        message: res.data.message
      });
      return;
    })
    .catch(async () => {
      await getBoards();
      await setStatus({
        open: true,
        type: 'error',
        message: 'ボードの削除に失敗しました。'
      });
      return;
    })
    .finally(() => {
      // スピナーoff
      setProgress(false);
      return;
    })
  }

  // ボードの削除時にstateから削除して、再度stateにセットし直している
  const deleteBoardState = (index: number) => {
    const newBoardsState = [ ...boardsState ];
    newBoardsState.splice(index, 1);
    setBoardsState(newBoardsState);
  }

  return (
    <ApiBoardsContext.Provider value={{
      boardsState,
      setBoardsState,
      getBoards,
      createBoard,
      updateBoard,
      updateBoardState,
      deleteBoard,
      deleteBoardState,
      }}
    >
      {props.children}
    </ApiBoardsContext.Provider>
  );
}