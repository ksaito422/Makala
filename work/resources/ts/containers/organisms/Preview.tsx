import React, { useContext } from 'react';
import MarkdownPreview from '@uiw/react-markdown-preview';
import { StylesContext } from '../../contexts/childContexts/StylesContext';
import {
  Paper
} from '@material-ui/core';

type props = {
  items: [
    {
      id: string,
      title: string,
      content: string,
    }
  ],
}

export const Preview: React.FC<props> = ({
  items
}) => {
  const { useStyles } = useContext<any>(StylesContext);
  const classes = useStyles();

  let previewText: string = '';
  const tmp = items.forEach(key => {
    previewText += key.title + key.content + '<br>'
  })

  return (
    <Paper elevation={3} className={classes.preview}>
      <MarkdownPreview source={previewText} />
    </ Paper>
  );
}