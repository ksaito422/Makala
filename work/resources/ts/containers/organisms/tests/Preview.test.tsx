import React from 'react';
import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import { Preview } from '../Preview';
import { StylesContextProvider } from '../../../contexts/childContexts/StylesContext';

// メモリリークを避けるために、各テスト後にすべてのコンテンツを消去する
afterEach(cleanup);

// プロパティの型定義
type Props = {
  items: [
    {
      id: string;
      title: string;
      content: string;
    }
  ];
};

describe('Preview', () => {
  it('スナップショットテスト', () => {
    const props: Props = {
      items: [
        {
          id: '1',
          title: 'string',
          content: 'string',
        },
      ],
    };

    const { baseElement } = render(
      <StylesContextProvider>
        <Preview {...props} />
      </StylesContextProvider>
    );
    expect(baseElement).toMatchSnapshot();
  });

  // it('ユニットテスト名', () => {
  //     ここに実行して欲しいテストを記述します。
  //   });
});
