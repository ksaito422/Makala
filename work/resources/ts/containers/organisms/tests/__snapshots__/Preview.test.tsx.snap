// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview スナップショットテスト 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root makeStyles-preview-8 MuiPaper-elevation3 MuiPaper-rounded"
          >
            <div
              class="wmde-markdown wmde-markdown-color "
            >
              <h1>
                string
              </h1>
              <p>
                string
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-centerPlacement-6 MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            class="MuiButtonBase-root MuiFab-root MuiFab-primary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiFab-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm0 15l-5-5h3V9h4v4h3l-5 5z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
